From f1b2ee2d8f5a7a0988e2794bae7350d2a9527cdb Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:23 -0400
Subject: [tools] perf: Reset output/sort order to default

Message-id: <1410181752-25631-259-git-send-email-jolsa@redhat.com>
Patchwork-id: 91944
O-Subject: [PATCH RHEL7.1 BZ1134356 258/307] perf tools: Reset output/sort order to default
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit d69b2962a0aebd431cdda939f4418dd606e2f77e
Author: Namhyung Kim <namhyung@kernel.org>
Date: Fri May 23 10:59:01 2014 +0900

description
===========
When reset_output_field() is called, also reset field/sort order to
NULL so that it can have the default values.  It's needed for testing.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/sort.c b/tools/perf/util/sort.c
index 9da8931..254f583 100644
--- a/tools/perf/util/sort.c
+++ b/tools/perf/util/sort.c
@@ -1582,6 +1582,9 @@ void reset_output_field(void)
  sort__has_sym = 0;
  sort__has_dso = 0;
 
+ field_order = NULL;
+ sort_order = NULL;
+
  reset_dimensions();
  perf_hpp__reset_output_field();
 }
-- 
1.7.1